homeassistant/components/backup/manager.py

```diff
@@ -240,18 +248,29 @@ def _mkdir_and_generate_backup_contents(
             tar_info.mtime = int(time.time())
             outer_secure_tarfile_tarfile.addfile(tar_info, fileobj=fileobj)
             with outer_secure_tarfile.create_inner_tar(
-                "./homeassistant.tar.gz", gzip=True
+                "./homeassistant.tar.gz",
+                gzip=True,
+                key=_password_to_key(password=password),
```

Comment:
This will only work if the backup is restored with a supervisor, as we do not have restore capability for core/container, Im unsure if this is good to add, as the files will not be restorable (even manually)

Reply:
Does that mean the only current use case would be to migrate from core to HAOS/supervised ?

Reply:
With the changes in https://github.com/home-assistant/core/pull/128365 this PR is now viable.